contrib/docker-compose/standalone/haproxy/etc/haproxy.cfg.tmpl

```diff
@@ -280,42 +269,101 @@ frontend fe_http_handler
     # Whitelisting options
     http-request allow if { req.hdr_ip(CF-Connecting-IP,-1) -f ${HAP_CONF_DIR}/lists/whitelist.list }
 
+    # Options
+    option httplog
+    option http-server-close
+    option dontlognull
+
+    log stdout format raw local0
+    log-format "%ci:%cp [%t] %ft %b/%s %Tq/%Tw/%Tc/%Tr/%Tt %ST %U/%B %CC %CS %tsc %ac/%fc/%bc/%sc/%rc %sq/%bq %hr %hs %{+Q}r"
+
+    # X-Forward- settings
+    http-request    set-header X-Forwarded-Proto http  if !{ ssl_fc }
+    http-request    set-header X-Forwarded-Proto https if { ssl_fc }
+    http-request    set-header X-Forwarded-Host  %[req.hdr(host)] if is_host_set
+
+    # HSTS response header
+    http-response   set-header Strict-Transport-Security "max-age=63072000; includeSubDomains; preload" if { ssl_fc }
+
+    # Secure response cookies
+    http-response   replace-header Set-Cookie ^((?:.(?!\ [Ss]ecure))*)$ \1;\ Secure if { ssl_fc }
+
+    # # Backend routing
+    use_backend be_acme_challenge  if acme_challenge !is_stopping
+    use_backend be_prometheus      if get_prometheus !is_stopping
+    #use_backend be_well_known_sec  if well_known_sec
+
+    default_backend be_hockeypuck_rewrite
+
+frontend fe_hockeypuck_ddos_lb
+    bind abns@hockeypuck-ddos accept-proxy
+    mode http
+
+    # Check for valid vhost names
+    acl keyserver_host      hdr_beg(host) -i ${FQDN}
+    #acl keyserver_host      hdr_beg(host) -i ${ALIAS_FQDN_1}
+    #acl keyserver_host      hdr_beg(host) -i ${ALIAS_FQDN_2}
+
+    # hockeypuck settings
```

Comment:
I just noticed that these acls are applied at the front end, whereas we do URL rewriting at the back end, so these won't match the alternative URLs (/s/ etc) ... so why do they still work?

Reply:
I have re-wired the solution. Now traffic flow is `fe_http_handler` -> `be_hockeypuck_rewrite` (where `/s/..` rewrite happens) -> `fe_fe_hockeypuck_ddos_lb` (where we check the re-written paths. In theory at this stage no `/s/...` call will make it as far) -> `primary` or `r/o backends`